src/utils/metrics.py

```diff
@@ -0,0 +1,156 @@
+            name="vllm:gpu_cache_usage_perc",
+            description="GPU KV-cache usage. 1 means 100 percent usage.",
+            kind=pb_utils.MetricFamily.GAUGE,
+        )
+        self.gauge_cpu_cache_usage_family = pb_utils.MetricFamily(
+            name="vllm:cpu_cache_usage_perc",
+            description="CPU KV-cache usage. 1 means 100 percent usage.",
+            kind=pb_utils.MetricFamily.GAUGE,
+        )
+
+        # Iteration stats
+        self.counter_num_preemption_family = pb_utils.MetricFamily(
+            name="vllm:num_preemptions_total",
+            description="Cumulative number of preemption from the engine.",
+            kind=pb_utils.MetricFamily.COUNTER,
+        )
+        self.counter_prompt_tokens_family = pb_utils.MetricFamily(
+            name="vllm:prompt_tokens_total",
+            description="Number of prefill tokens processed.",
+            kind=pb_utils.MetricFamily.COUNTER,
+        )
+        self.counter_generation_tokens_family = pb_utils.MetricFamily(
+            name="vllm:generation_tokens_total",
+            description="Number of generation tokens processed.",
+            kind=pb_utils.MetricFamily.COUNTER,
+        )
+
+        # System stats
+        #   Scheduler State
+        self.gauge_scheduler_running = self.gauge_scheduler_running_family.Metric(
+            labels=labels
+        )
+        self.gauge_scheduler_waiting = self.gauge_scheduler_waiting_family.Metric(
+            labels=labels
+        )
+        self.gauge_scheduler_swapped = self.gauge_scheduler_swapped_family.Metric(
+            labels=labels
+        )
+        #   KV Cache Usage in %
+        self.gauge_gpu_cache_usage = self.gauge_gpu_cache_usage_family.Metric(
+            labels=labels
+        )
+        self.gauge_cpu_cache_usage = self.gauge_cpu_cache_usage_family.Metric(
+            labels=labels
+        )
+
+        # Iteration stats
+        self.counter_num_preemption = self.counter_num_preemption_family.Metric(
+            labels=labels
+        )
+        self.counter_prompt_tokens = self.counter_prompt_tokens_family.Metric(
+            labels=labels
+        )
+        self.counter_generation_tokens = self.counter_generation_tokens_family.Metric(
+            labels=labels
+        )
+
+
+class VllmStatLogger(VllmStatLoggerBase):
+    """StatLogger is used as an adapter between vLLM stats collector and Triton metrics provider."""
+
+    # local_interval not used here. It's for vLLM logs to stdout.
+    def __init__(self, labels: Dict, local_interval: float = 0) -> None:
+        # Tracked stats over current local logging interval.
+        super().__init__(local_interval)
+        self.metrics = TritonMetrics(labels=labels)
+
+    def info(self, type: str, obj: SupportsMetricsInfo) -> None:
+        raise NotImplementedError
+
+    def _log_gauge(self, gauge, data: Union[int, float]) -> None:
+        # Convenience function for logging to gauge.
```

Comment:
How about we convert the description to a Google docstring format : https://github.com/google/styleguide/blob/gh-pages/pyguide.md#383-functions-and-methods .

Reply:
Updated